src/main/java/friday/task/Task.java

```diff
@@ -0,0 +1,60 @@
+package friday.task;
+
+/**
+ * Represents an abstract Task. A Task can be marked as done or not done and has a description.
+ * This class serves as a base class for specific types of tasks like Todo, Deadline, and Event.
+ */
+public abstract class Task {
+    private String description;
+    private boolean isDone;
+
+    /**
+     * Constructs a Task with the specified description and completion status.
+     *
+     * @param description The description of the Task.
+     * @param isDone      True if the task is done, false otherwise.
+     */
+    public Task(String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    /**
+     * Marks the Task as done.
+     */
+    public void markAsDone() {
```

Comment:
Should this be setDone instead to fulfill the method naming criteria for setters?